### PR TITLE
Rdiaz82/task/update tsc version

### PR DESCRIPTION
**Pull request by rdiaz82**
## Description
Updated the version of tsc to the latest version and fixed new check types for exceptions. 

## Changes

updated `package.json` and casted all exceptions 

## Checks
- [x] Project Builds
- [x] Project passes tests and checks
- [ ] Updated documentation accordingly

### CODE REVIEW

File: packages/cli/src/common/script.ts [hunk 1]
```diff
@@ -126,7 +126,8 @@ export class Script<TContext> {
           throw err
         }, constVoid)
       )
-    } catch (err) {
+    } catch (error) {
+      const err = error as Error
```
Comment (javiertoledo) [inline packages/cli/src/common/script.ts:130]: If this is now required, I wonder if they enabled type annotations in the catch error type:

```suggestion
    } catch (error: Error) {
```

File: packages/framework-core/src/booster-graphql-dispatcher.ts [hunk 2]
```diff
@@ -66,7 +66,8 @@ export class BoosterGraphQLDispatcher {
       try {
         this.logger.debug(`Decoding current user from auth token: ${envelope.token}`)
         envelope.currentUser = await this.boosterTokenVerifier.verify(envelope.token)
-      } catch (e) {
+      } catch (err) {
+        const e = err as string
```
Comment (javiertoledo) [inline packages/framework-core/src/booster-graphql-dispatcher.ts:70]: 🤔  Is this intended? Even while you can throw a string, I think that the language wraps it in an Error object...

**Review by javiertoledo — APPROVED** [review]
I guess that this is what's needed, but it looks weird to me that we need those casts, I've left a couple of comments just in case there's an alternative. Otherwise, I'm ok with this.

File: packages/cli/src/common/script.ts (hunk 1 above)
Comment (rdiaz82) [reply packages/cli/src/common/script.ts:130]: yeahh!! that what I expected but not! In fact, it is even weirder you can use `error:any` but you can't use any other type. This is what the official documentation says:

https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/#use-unknown-catch-variables

File: packages/framework-core/src/booster-graphql-dispatcher.ts (hunk 2 above)
Comment (rdiaz82) [reply packages/framework-core/src/booster-graphql-dispatcher.ts:70]: not now! now errors are `unkown` and you are forced to define the type. In that particular case the error was a string.

Comment (javiertoledo) [reply packages/framework-core/src/booster-graphql-dispatcher.ts:70]: That's weird, maybe we should try to fix that and make sure errors are always wrapped in an `Error` object 🤔

Comment (rdiaz82) [reply packages/framework-core/src/booster-graphql-dispatcher.ts]: Fixed!! I checked and the function throwing the error was generating a normal `Error` so simply was a matter of changing the streaming error from simply pass the error to `err.message`

**Comment by rdiaz82** [comment]
This was fixed in anothe PR adding a flag, so closing this one!

**Pull request closed without merging**